### PR TITLE
Update asv config + fix some broken benchmarks

### PR DESCRIPTION
**Pull request by pv**
- Enable platform-dependent config in asv (needs asv git version for it to do something)
- Enable wheel cache in asv (in asv git version)
- Fix a few easily fixed broken benchmarks

### CODE REVIEW

File: asv_bench/benchmarks/eval.py [hunk 1]
```diff
@@ -3,192 +3,36 @@
...
-    def setup(self):
+    def setup(self, engine, threads):
         self.df = DataFrame(np.random.randn(20000, 100))
         self.df2 = DataFrame(np.random.randn(20000, 100))
         self.df3 = DataFrame(np.random.randn(20000, 100))
         self.df4 = DataFrame(np.random.randn(20000, 100))
 
-    def time_eval_frame_chained_cmp_python(self):
-        pd.eval('df < df2 < df3 < df4', engine='python')
+        if threads == 1:
```
Comment (jreback) [inline asv_bench/benchmarks/eval.py]: hmm maybe should set this explicity each time something is run to 1 (then you can set in this particular one to higher)

Comment (pv) [inline asv_bench/benchmarks/eval.py]: I'm not sure what would be a good value for threads=='all', could you explain in code how you'd like it. Note each benchmark is run in its own process, so those with threads!=1 get the default value on pandas startup.

Comment (jreback) [inline asv_bench/benchmarks/eval.py]: oh, each is in own thread, then ok. I just didn't want to set a 'global', but if spinning off proc then its fine.

**Comment by jreback** [comment]
can you rebase this. we can prob throw this in 0.18.0; it doesn't affect user code.

Can you run vs 0.17.1 and post anything interesting?

**Comment by pv** [comment]
Rebased.
Don't have time to run the full bench suite right now, but can be done via
`asv continuous 6c30cbecf8 master`

**Comment by pv** [comment]
benchmarks: https://gist.github.com/pv/8ff97c3f6faa62fd6a5e --- not sure if all of those are real, or fluctuations (laptop under load)

**Comment by jreback** [comment]
I think we need to update the perf docs in contributing?

**Comment by pv** [comment]
Not sure --- they already instruct to install the asv git version.

File: asv_bench/asv.conf.json [hunk 2]
```diff
@@ -30,24 +30,62 @@
...
+    // key-value pairs to include/exclude.
+    //
+    // An exclude entry excludes entries where all values match. The
+    // values are regexps that should match the whole string.
+    //
+    // An include entry adds an environment. Only the packages listed
+    // are installed. The 'python' key is required. The exclude rules
+    // do not apply to includes.
+    //
+    // In addition to package names, the following keys are available:
```
Comment (jreback) [inline asv_bench/asv.conf.json]: so how does one choose the installation method / use this, ideally would ha e examples for conda (default) / pip

**Comment by pv** [comment]
Added explanation on how to choose between conda / virtualenv / existing environment to contributer docs.

**Comment by jreback** [comment]
thanks @pv 

pls check the docs when built (prob a few hours) and if necessary issue a follow-up

**Comment by jreback** [comment]
should these be ignored as well?
```
[jreback-~/pandas] git st
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        asv_bench/.nojekyll
        asv_bench/results.orig/
        asv_bench/run_versions.sh

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by jreback** [comment]
nvm mind the ``results.orig``.......

**Pull request closed without merging**